pennylane/transforms/decompositions/clifford_t/clifford_t_transform.py

```diff
@@ -0,0 +1,479 @@
+
+
+def check_clifford_op(op):
+    r"""Checks if an operator is Clifford or not.
+
+    For a given unitary operator :math:`U` acting on :math:`N` qubits, this method checks that the
+    transformation :math:`UPU^{\dagger}` maps the Pauli tensor products :math:`P = {I, X, Y, Z}^{\otimes N}`
+    to Pauli tensor products with :math:`O(N \times 8^N)` time complexity when using naive matrix multiplication.
+
+    Args:
+        op: the operator that needs to be tested
+
+    Returns:
+        Bool that represents whether the provided operator is Clifford or not.
+    """
+
+    # Check if matrix can be calculated for the operator
+    if not op.has_matrix:
+        return False
+
+    # Compute the LCUs for the operator in Pauli basis
+    pauli_terms = qml.pauli_decompose(qml.matrix(op), wire_order=op.wires, check_hermitian=False)
+    pauli_terms_adj = qml.Hamiltonian(qml.math.conj(pauli_terms.coeffs), pauli_terms.ops)
+
+    # Build Pauli tensor products P in Hamiltonian form
+    def pauli_group(x):
+        return [qml.Identity(x), qml.PauliX(x), qml.PauliY(x), qml.PauliZ(x)]
+
+    # Build PauliSentence and Hamiltonians representations for set P
+    pauli_sens = [
+        qml.pauli.pauli_sentence(qml.prod(*pauli))
+        for pauli in product(*(pauli_group(idx) for idx in op.wires))
+    ]
+    pauli_hams = (pauli_sen.hamiltonian(wire_order=op.wires) for pauli_sen in pauli_sens)
+
+    # Perform U@P@U^\dagger and check if the result exists in set P
+    for pauli_prod in product([pauli_terms], pauli_hams, [pauli_terms_adj]):
+        # hopefully op_math.prod scales better than matrix multiplication, i.e., O((2^N)^3)
+        upu = qml.pauli.pauli_sentence(qml.prod(*pauli_prod))
+        upu.simplify()
+
+        if len(upu) != 1:  # Pauli sum always lie outside set P
+            return False
+
+    return True
+
+
+def check_clifford_t(op):
+    r"""Checks whether the gate is in the standard Clifford+T basis.
+
+    For a given unitary operator :math:`U` acting on :math:`N` qubits, which is not a T-gate,
+    this method checks that the transformation :math:`UPU^{\dagger}` maps the Pauli tensor products
+    :math:`P = {I, X, Y, Z}^{\otimes N}` to Pauli tensor products with :math:`O(N \times 8^N)` time
+    complexity when using naive matrix multiplication.
+
+    Args:
+        op: the operator that needs to be checked
+
+    Returns:
+        Bool that represents whether the provided operator is Clifford+T or not.
+    """
+    # Get the base operation for an adjointed operation
+    base = getattr(op, "base", None)
+
+    # Save time and check from the pre-computed list
+    if isinstance(op, _CLIFFORD_T_GATES) or isinstance(base, _CLIFFORD_T_GATES):
+        return True
+
+    # Save time and check from the parameter of rotation gates
+    if isinstance(op, _PARAMETER_GATES) or isinstance(base, _PARAMETER_GATES):
+        return qml.math.isclose(op.data[0] % math.pi, 0.0)
```

Comment:
Are we testing for each element of `_PARAMETER_GATES`? This one worries me about because it assumes information about how gate parameters are treated (e.g., divided by two or not?)

Reply:
Yes. Is the concern regarding jitting this? 🤔

Reply:
The parameterized gates that we use specifically for this transform are periodic and follow these simplifications. Added a note regarding it.

Reply:
Resolution - we are testing each one, so this is fine